source/lac/petsc_precondition.cc

```diff
@@ -479,6 +479,8 @@ namespace PETScWrappers
   void
   PreconditionBoomerAMG::initialize ()
   {
+
+#ifdef PETSC_HAVE_HYPRE
 #ifndef PETSC_USE_COMPLEX
```

Comment:
i think you can remove the `PETSC_USE_COMPLEX` guard and just use the one you introduce `PETSC_HAVE_HYPRE`. Should be equivalent. But it would be good to check that it still compiles with complex PETSc.